sharding/sharding.go

```diff
@@ -189,12 +199,14 @@ func (r *ShardingFerry) Run() {
 	r.Ferry.FlushBinlogAndStopStreaming()
 	copyWG.Wait()
 
+	// Joined tables cannot be easily monitored for changes in the binlog stream
```

Comment:
Curious as to why joined tables create this problem. Is there a related issue?

Reply:
Joined tables usually don't have the sharding key stored on them (because a single row can be shared by multiple tenants), hence discovering applicable (i.e. belonging to the tenant) DML events produced by these tables is impossible. 

Note: you could technically do the join every time you see a DML event and _find out_ if it's applicable but this could really slow down the replication process for Ghostferry